examples/manifold/plot_t_sne_perplexity.py

```diff
@@ -0,0 +1,104 @@
+"""
+=============================================================================
+ t-SNE: The effect of various perplexity values on the shape
+=============================================================================
+
+An illustration of t-SNE on the two concentric circles and the S-curve
+datasets for different perplexity values.
+
+We observe the tendency of higher variations for small perplexity values and
+clearer shapes as the value increases.
+
+The size, the distance and the shape of clusters may vary upon initialization,
+perplexity values and does not always convey a meaning.
+
+As shown below, t-SNE for higher perplexities finds meaningful topology of
+two concentric circles, however the size and the distance of the circles varies
+slightly from the original. In contrary to two circles dataset, the shapes
+visually diverge from S-curve topology on S-curve dateset even for larger
+perplexity values.
+
+For further details, "How to Use t-SNE Effectively"
+http://distill.pub/2016/misread-tsne/ provides a good discussion of the
+effects of various parameters, as well as interactive plots to explore
+those effects.
+"""
+
+# License: BSD 3 clause
+
+print(__doc__)
+
+import matplotlib.pyplot as plt
+
+from matplotlib.ticker import NullFormatter
+from mpl_toolkits.mplot3d import Axes3D
+from sklearn import manifold, datasets
+from time import time
+
+# Next line to silence pyflakes. This import is needed.
```

Comment:
Removed! 
Now is is failing on ci tests:
'''
Exception occurred:
  File "/home/ubuntu/scikit-learn/doc/sphinxext/sphinx_gallery/gen_gallery.py", line 280, in sumarize_failing_examples
    "\n" + "-" * 79)
ValueError: Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
/home/ubuntu/scikit-learn/examples/applications/plot_stock_market.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/ubuntu/scikit-learn/doc/sphinxext/sphinx_gallery/gen_rst.py", line 472, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 104, in <module>
  File "<string>", line 104, in <listcomp>
  File "<string>", line 35, in quotes_historical_google
  File "/home/ubuntu/miniconda/envs/testenv/lib/python3.6/site-packages/numpy/lib/npyio.py", line 1775, in genfromtxt
    converters[i].update(conv, locked=True,
IndexError: list index out of range
'''
I'm not sure if the error is related to my changes